src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -140,6 +171,9 @@ private bool AddCommonHeaders(RequestContext requestContext, bool addCorrelation
 
             _headers.Add(TelemetryConstants.XClientLastTelemetry, _telemetryManager.FetchAndResetPreviousHttpTelemetryContent());
             _headers.Add(TelemetryConstants.XClientCurrentTelemetry, _telemetryManager.FetchCurrentHttpTelemetryContent(requestContext.CorrelationId.AsMatsCorrelationId()));
+
+            //add pkeyauth header
+            _headers.Add(DeviceAuthHeaderName, DeviceAuthHeaderValue);
```

Comment:
So we add this on all platforms, even if a deviceManager does not exist? I would argue that if a device manager does not  exist, then we do not add the header? 

From an implementation detail, I would let the deviceManager itself add this header, that way the NullDeviceManager would not nothing, while the other managers can add it.